### PR TITLE
K8SPXC-263 Change status host if proxysql type LoadBalancer

### PR DESCRIPTION
**Pull request by nonemax**

### CODE REVIEW

File: pkg/controller/pxc/status.go [hunk 1]
```diff
@@ -103,7 +104,23 @@ func (r *ReconcilePerconaXtraDBCluster) updateStatus(cr *api.PerconaXtraDBCluste
...
+			err := r.client.Get(context.TODO(),
+				types.NamespacedName{
+					Namespace: cr.Namespace,
+					Name:      cr.Name + "-" + "proxysql",
+				}, svc)
+			if err != nil {
+				return errors.Wrap(err, "get proxysql service")
+			}
+			for _, i := range svc.Status.LoadBalancer.Ingress {
+				cr.Status.Host = i.IP
```
Comment (delgod) [inline pkg/controller/pxc/status.go:120]: sometimes it is IP, sometimes it is Hostname, also it takes time to get it from Ingress
please take function from PSMDB Operator https://github.com/percona/percona-server-mongodb-operator/blob/master/pkg/psmdb/service.go#L148-L185

File: pkg/controller/pxc/status.go [hunk 2]
```diff
@@ -103,7 +104,23 @@ func (r *ReconcilePerconaXtraDBCluster) updateStatus(cr *api.PerconaXtraDBCluste
 		}
 
 		cr.Status.ProxySQL = proxyStatus
-		cr.Status.Host = cr.Name + "-" + "proxysql"
+
+		cr.Status.Host = cr.Name + "-" + "proxysql." + cr.Namespace
```
Comment (delgod) [inline pkg/controller/pxc/status.go:108]: what if proxysql disabled?

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| recreate | passed |
| storage | passed |
| big-data | failed |
| init-deploy | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| limits | passed |
| upgrade | passed |
| restore-to-encrypted-cluster | failed |
| monitoring | passed |
| self-healing-advanced | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| upgrade-consistency | passed |
| security-context | passed |
| operator-self-healing | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| scheduled-backup | passed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | failed |

File: pkg/controller/pxc/status.go (hunk 1 above)
Comment (nonemax) [reply pkg/controller/pxc/status.go:120]: added https://github.com/percona/percona-xtradb-cluster-operator/pull/373/commits/a3672fcc146c5e81a5adabd0d109b43771dfe922

File: pkg/controller/pxc/status.go (hunk 2 above)
Comment (nonemax) [reply pkg/controller/pxc/status.go:108]: https://github.com/percona/percona-xtradb-cluster-operator/pull/373/commits/a3672fcc146c5e81a5adabd0d109b43771dfe922

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| big-data | passed |
| scaling | passed |
| storage | failed |
| recreate | passed |
| scaling-proxysql | passed |
| limits | passed |
| restore-to-encrypted-cluster | failed |
| monitoring | passed |
| upgrade | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| upgrade-consistency | passed |
| security-context | passed |
| affinity | passed |
| self-healing-advanced | passed |
| one-pod | passed |
| auto-tuning | passed |
| scheduled-backup | passed |
| operator-self-healing | failed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| recreate | passed |
| scaling | passed |
| restore-to-encrypted-cluster | failed |
| init-deploy | passed |
| scaling-proxysql | passed |
| limits | passed |
| upgrade | passed |
| monitoring | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| security-context | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| big-data | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-encrypted-with-tls | failed |

File: pkg/controller/pxc/status.go (hunk 1 above)
Comment (dAdAbird) [reply pkg/controller/pxc/status.go:120]: But there is still no workaround for ingress lags, innit?

File: pkg/controller/pxc/status.go [hunk 3]
```diff
@@ -103,7 +104,26 @@ func (r *ReconcilePerconaXtraDBCluster) updateStatus(cr *api.PerconaXtraDBCluste
...
+		if cr.Spec.ProxySQL.ServiceType != nil && *cr.Spec.ProxySQL.ServiceType == corev1.ServiceTypeLoadBalancer {
```
Comment (dAdAbird) [inline pkg/controller/pxc/status.go:109]: Not mandatory, but consider changing cr.Spec.ProxySQL.ServiceType to the non-pointer type. There is no reason for it to be a pointer. We always dereference it thru all the code.

Comment (nonemax) [reply pkg/controller/pxc/status.go]: https://github.com/percona/percona-xtradb-cluster-operator/pull/373/commits/d6e19c57762498e95ec902e214a382db7af0e979

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| recreate | passed |
| init-deploy | failed |
| storage | passed |
| scaling | passed |
| limits | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| monitoring | failed |
| restore-to-encrypted-cluster | failed |
| upgrade | passed |
| monitoring-2-0 | failed |
| demand-backup | failed |
| affinity | passed |
| self-healing-advanced | passed |
| upgrade-consistency | passed |
| security-context | passed |
| operator-self-healing | passed |
| scheduled-backup | failed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | failed |
| recreate | passed |
| restore-to-encrypted-cluster | failed |
| storage | passed |
| big-data | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| scaling | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| security-context | passed |
| limits | passed |
| monitoring | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| scheduled-backup | failed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| recreate | passed |
| storage | failed |
| scaling | passed |
| big-data | passed |
| scaling-proxysql | passed |
| limits | passed |
| upgrade | passed |
| restore-to-encrypted-cluster | failed |
| monitoring | passed |
| self-healing | passed |
| upgrade-consistency | passed |
| security-context | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| self-healing-advanced | passed |
| affinity | passed |
| operator-self-healing | passed |
| one-pod | failed |
| auto-tuning | passed |
| scheduled-backup | passed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | failed |
| big-data | passed |
| scaling | passed |
| storage | passed |
| recreate | passed |
| limits | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| upgrade | passed |
| restore-to-encrypted-cluster | failed |
| monitoring | passed |
| demand-backup | passed |
| self-healing-advanced | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| security-context | passed |
| operator-self-healing | passed |
| affinity | passed |
| one-pod | passed |
| scheduled-backup | failed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | failed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| init-deploy | passed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | failed |
| storage | passed |
| scaling | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| security-context | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-encrypted-with-tls | passed |

**Comment by JNKPercona** [comment]
PXC operator docker - perconalab/percona-xtradb-cluster-operator:PR-373-bdf417b4

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| storage | passed |
| scheduled-backup | passed |
| self-healing | passed |
| demand-backup-encrypted-with-tls | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| scaling | passed |
| big-data | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| security-context | passed |
| init-deploy | passed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |

**Pull request merged**